api/routes/login.js

```diff
@@ -1,69 +1,105 @@
 const express = require('express');
+const bodyparser = require('body-parser');
+const cookieParser = require('cookie-parser');
+const User = require('../models/user');
+const { auth } = require('../middleware');
 const path = require('path');
-const jwt = require('jsonwebtoken');
-const withAuth = require('../middleware');
-const User = require('../models/User');
 
-const {JWT_SECRET} = require('../config');
 
 const app = express();
-
-app.get('/', function (req, res) {
-  res.sendFile(path.join(__dirname, 'public', 'index.html'));
+// app use
+app.use(bodyparser.urlencoded({ extended: false }));
+app.use(bodyparser.json());
+app.use(cookieParser());
+
+app.use('/', express.static(path.join(__dirname, 'static')));
+
+// adding new user (sign-up route)
+app.post('/register', function(req, res) {
+    let newUser = new User(req.body);
+    User.findOne({ email: newUser.email }, function(err, user) {
+        if (user) return res.status(400).json({
+            auth: false,
+            message: "email exits"
+        });
+
+        newUser.save((err, doc) => {
+            if (err) {
+                return res.status(400).json({
+                    error: true,
+                    message: err.message
+                });
+            }
+            res.status(200).json({
+                success: true,
+                user: { // Control the output we don't want to return the password that would be bad
```

Comment:
FIX: this was an important change NEVER expose the password to the frontend. Even though it is hashed this poses a security risk. We make it way harder for the hacker to get the hashes if he needs access to the server db.